test/assignment3/Assignment3Test.java

```diff
@@ -0,0 +1,81 @@
+package assignment3;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class Assignment3Test {
+    
+    private static final int HEIGHT = 2;
+    private static final int WIDTH = 3;
+    private static final char[][] GRID = new char[HEIGHT][WIDTH];
+    
+    @BeforeClass
+    public static void setUpClass() {

```

Comment:
It is better to fully initialize final fields immediately when they are declared. This makes it clear, what value they have.

`private static final char[][] GRID = {{'a', 'a', 'r'}, {'t', 'c', 'd'}};`